### ISSUE TITLE
Building HPX with support for PAPI fails

### ISSUE THREAD

**Issue by kordejong** [STEllAR-GROUP/hpx#4755]
## Expected Behavior

Build succeeds

## Actual Behavior

Builds fails. Error message:

```
/tmp/hello_hpx-fetch_content/RelWithDebInfo-papi/build/_deps/hpx-src/components/performance_counters/papi/src/papi_startup.cpp: In function ‘void hpx::performance_counters::papi::find_threads(std::set<std::__cxx11::basic_string<char> >&, hpx::performance_counters::discover_counters_mode)’:
/tmp/hello_hpx-fetch_content/RelWithDebInfo-papi/build/_deps/hpx-src/components/performance_counters/papi/src/papi_startup.cpp:110:51: error: invalid use of incomplete type ‘class hpx::runtime’
         hpx::util::thread_mapper& tm = get_runtime().get_thread_mapper();
```

[build_output.txt](https://github.com/STEllAR-GROUP/hpx/files/4779144/build_output.txt)

## Steps to Reproduce the Problem

  1. Build HPX with support for PAPI. My script used this command to configure HPX: `cmake -DCMAKE_BUILD_TYPE=/tmp/hello_hpx-fetch_content/RelWithDebInfo-papi -DHELLO_HPX_REPOSITORY_CACHE:PATH=/home/kor/development/repository papi -DHPX_WITH_PAPI:BOOL=ON -HPX_WITH_MALLOC:STRING=tcmalloc -HPX_WITH_HWLOC:BOOL=ON -S /home/kor/development/project/hello_hpx -B /tmp/hello_hpx-fetch_content/RelWithDebInfo-papi/build`
  1. Build HPX
  1. Wait for error message. Occurs at 61% of the build in my case.

## Specifications

  - HPX Version: master of last Friday (commit 47d7bebd27a)
  - Platform (compiler, OS): gcc-7, Kubuntu 18.04

**Comment by msimberg** [comment]
Sorry about this, we'll fix this asap.

**Comment by kordejong** [comment]
No problem, take your time ;-) I just found this when running my build test scripts. I don't use this specific build at the moment. (Some fix for issue #4746 is actually much more important for me right now.)
